public/js/controller.js

```diff
@@ -73,21 +72,55 @@ function rgba_to_grayscale(rgba, nrows, ncols) {
     return gray;
 }
 
+/**
+ * Runs the model on a single frame.
+ * @param {Object} video                    Video object.
+ * @param {Array} detectResult              Array containing results from face detection model.
+ * @return {Int32Array}                     Returns the results from the model in an array.
+ */
+async function runPrediction(video, detectResult) {
+    let ppl_count = 0;
+    let input = new Array()
+
+    for (let i = 0; i < detectResult.length; i++) {
+        ppl_count += 1
+        let x = detectResult[i][0];
+        let y = detectResult[i][1];
+        let bbxwidth = detectResult[i][2];
+        let bbxheight = detectResult[i][3];
+
+        let cropImage = document.createElement('canvas')
+        cropImage.width = bbxwidth
+        cropImage.height = bbxheight
+        let cropCtx = cropImage.getContext('2d')
+        cropCtx.drawImage(video, x, y, bbxwidth, bbxheight, 0, 0, bbxwidth, bbxheight)
+        input.push(cropImage)
+    }
+
+    // Inference
+    if (input.length > 0) {
+        return ageEstimator.predict(input)
+    } else {
+        return Promise.resolve()
+    }
+}
+
 /**
  * Takes in an array of bounding boxes and ages and updates the frame with the results.
  * @param {Array[Array[Number]]} faces          Array containing an array of numbers in the format [x, y, width, height]
  * @param {Array[Number]} ageResult             Array containing ages
  */
 function processAgeResults(faces, ageResult) {
-    for (i = 0; i < faces.length; ++i) {
+    for (i = 0; (i < faces.length && ageResult); ++i) {
+        console.log(ageResult);
```

Comment:
maybe comment out or remove the console logging before merging?

Reply:
Sure, I left it in because I was still testing a few things.